### PR TITLE
LPS-150420 Audit and refactor tests on WebContent | master

### PR DESCRIPTION
**Pull request by ozysouza**
https://issues.liferay.com/browse/LPS-150420

cc @david-gutierrez-mesa 
cc @Tim-Cao

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentStructures.testcase [hunk 1]
```diff
@@ -206,51 +226,42 @@ definition {
...
-		property testray.component.names = "Web Content Administration";
-
-		WebContentNavigator.openWebContentAdmin(siteURLKey = "test-site-name");
-
-		NavItem.gotoStructures();
-
-		WebContentStructures.addCP(
-			structureDescription = "WC Structure Description",
-			structureName = "WC Structure 1");
+		task ("Given a Web Content admin adds a new structure") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentStructures.testcase:232]: Given a Web Content admin has a structure based on a fieldset

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentStructures.testcase [hunk 2]
```diff
@@ -206,51 +226,42 @@ definition {
...
+			DataEngine.addFields(fieldNameList = "Text,Rich Text,Image,Upload");
 
-		DataEngine.addFieldsetToBuilder(fieldsetName = "WC Structure 1");
+			WebContentStructures.saveCP();
+		}
 
-		DataEngine.addField(
-			fieldFieldLabel = "Numeric",
-			fieldName = "Numeric");
+		task ("When the Web Content admin adds a new structure with Numeric field and fieldsets from first structure") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentStructures.testcase:244]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentStructures.testcase [hunk 3]
```diff
@@ -206,51 +226,42 @@ definition {
...
+			DataEngine.addField(
+				fieldFieldLabel = "Numeric",
+				fieldName = "Numeric");
 
-		NavItem.gotoStructures();
+			WebContentStructures.saveCP();
+		}
 
-		WebContentStructures.editStructureDefaultValuesCP(structureName = "WC Structure 2");
+		task ("Then the Web Content admin can edit the default values of second structure") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentStructures.testcase:258]: When the Web Content admin defines the default values of structure

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentStructures.testcase [hunk 4]
```diff
@@ -206,51 +226,42 @@ definition {
...
+			WebContentStructures.editStructureDefaultValuesCP(structureName = "WC Structure 2");
 
-		WebContent.addWithStructureCP(webContentText = "Edited Text Field");
+			WebContent.addWithStructureCP(webContentText = "Edited Text Field");
 
-		PortletEntry.save();
+			PortletEntry.save();
+		}
 	}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentStructures.testcase:265]: Add step **Then the default values of structure should be persisted**

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentStructures.testcase [hunk 5]
```diff
@@ -206,51 +226,42 @@ definition {
 		}
 	}
 
+	@description = "The user cand edit the default values for a structure with fieldsets"
 	@priority = "4"
 	test EditStructureDefaultValuesForFieldsets {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentStructures.testcase:231]: EditStructureDefaultValuesBasedOnFieldset

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentStructures.testcase [hunk 6]
```diff
@@ -605,48 +616,52 @@ definition {
...
 
+	@description = "Validate global structure and template."
 	@priority = "4"
-	test ValidateDDMNavigation {
-		property test.name.skip.portal.instance = "WebContentStructures#ValidateDDMNavigation";
-		property testray.component.names = "Web Content Administration";
-
-		SearchAdministration.executeReindex();
+	test ValidateGlobalTemplatesForBasicWebContent {
+		task ("Given a Web Content admin adds a new template for Basic Web Content in global site") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentStructures.testcase:622]: Given a Web Content admin has a new template for Basic Web Content in global site

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentStructures.testcase [hunk 7]
```diff
@@ -605,48 +616,52 @@ definition {
...
+			Navigator.gotoBack();
+		}
 
-		SiteAdmin.viewPortletTitle(portletTitle = "Templates for Structure: Basic Web Content");
+		task ("When the Web Content admin changes the display style to table") {
+			LexiconEntry.changeDisplayStyle(displayStyle = "table");
+		}
 
-		LexiconEntry.changeDisplayStyle(displayStyle = "table");
+		task ("Then the new web conten template and default are shown") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentStructures.testcase:642]: Then the templates are shown as table entry

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentStructures.testcase [hunk 8]
```diff
@@ -704,49 +719,50 @@ definition {
 		}
 	}
 
+	@description = "The user can add a favorite structure in web content admin."
 	@priority = "4"
-	test ViewFavoriteStructure {
-		property testray.component.names = "Web Content Administration";
+	test ViewFavoriteStructureInWebContentAdmin {
+		task ("Given a Web Content admin adds seven new structures with text field") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentStructures.testcase:725]: Given a Web Content admin has multiple structures

**Review by Tim-Cao — COMMENTED** [review]
Hi @ozysouza, could you modify according to comments? Thanks

**Comment by ozysouza** [comment]
> Hi @ozysouza, could you modify according to comments? Thanks

Hi @Tim-Cao, PR updated. Would you mind have a look at? Thanks!

**Comment by Tim-Cao** [comment]
Good job! LGTM! @ozysouza

**Comment by Tim-Cao** [comment]
ci:forward

**Pull request closed without merging**